### ISSUE TITLE
Pretraining VisualBert using custom dataset

### ISSUE THREAD

**Issue by monkeysforever** [facebookresearch/mmf#515]
## ❓ Questions and Help
I want to pretrain VisualBert pretrained on COCO dataset on a domain specific custom dataset. Any help regarding how I should process would be most appreciated!

**Comment by apsdehal** [comment]
Hi,

when you say you want to pretrain VisualBERT already pretrained on COCO do you want to do fine-tuning on a dataset or continue pretraining?

**Comment by monkeysforever** [comment]
Pretrain it further on a dataset relevant to the downstream task i want to finetune on.

**Comment by apsdehal** [comment]
VisualBERT works on region features. So, you have to follow the steps below to achieve the further pretraining given that your images are new:
- Extract out the region features for your own dataset by using feature extraction script: https://github.com/facebookresearch/mmf/blob/master/tools/scripts/features/extract_features_vmb.py. Let's say you extracted your features in the folder x in the format: train: x/train, val: x/val, test: x/test
- Next step is to convert your annotations to format of COCO dataset so that it works outside the box. Have a look at how the annotations looks for coco npy format by using `np.load("path_to_coco_annotation_db.npy", allow_pickle=True)[1]`.
- Update the visual bert config at https://github.com/facebookresearch/mmf/blob/master/projects/visual_bert/configs/masked_coco/defaults.yaml to include updated dataset_config following https://github.com/facebookresearch/mmf/blob/master/mmf/configs/datasets/coco/masked.yaml
- Run the command for pretraining while appending `checkpoint.resume_zoo=visual_bert.pretrained.coco.full`.

Let me know if you get stuck at any stage or if something is not clear.

**Comment by monkeysforever** [comment]
Thanks
